snakeoil/test/test_osutils.py

```diff
@@ -311,44 +312,127 @@ def test_nonexistent(self):
         self.assertRaises(osutils.NonExistent, osutils.FsLock,
                           pjoin(self.dir, 'missing'))
 
-    def test_locking(self):
-        path = pjoin(self.dir, 'lockfile')
+    @staticmethod
```

Comment:
Just a thought.  Any reason you don't just convert this into a decorator?  You're chunking up some tests into "in process" and "out of process"... I'm probably missing something, but is there any reason you can't just push it all into a subprocess?

The benefit of doing this is that if you used a decorator instead, you could enforce timeouts- so if the test fails, it doesn't hang indefinitely.